libraries/standard/http/include/http_client.h

```diff
@@ -1,347 +1,434 @@
+         * will send the headers in the buffer first, then fill the buffer with
+         * the response message.
+         */
+        uint8_t * pBuffer;
+        size_t bufferLen; /**< The length of the response buffer in bytes. */
+
+        /**
+         * @brief Optional callback for intercepting the header during the first
+         * parse through of the response as is it receive from the network.
+         * Set to NULL to disable.
+         */
+        HTTPClient_HeaderParsingCallback_t * pHeaderParsingCallback;
+
+        /**
+         * @brief The starting location of the response headers in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint8_t * pHeaders;
+
+        /**
+         * @brief Byte length of the response headers in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t headersLen;
+
+        /**
+         * @brief The starting location of the response body in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint8_t * pBody;
+
+        /**
+         * @brief Byte length of the body in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t bodyLen;
+
+        /* Useful HTTP header values found. */
+
+        /**
+         * @brief The HTTP response Status-Code.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint16_t statusCode;
+
+        /**
+         * @brief The value in the "Content-Length" header is returned here.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t contentLength;
+
+        /**
+         * @brief Count of the headers sent by the server.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t headerCount;
+
+        /**
+         * @brief Flags of useful headers found in the response.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint32_t flags;
+    } HTTPResponse_t;
 
 /**
  * @brief Initialize the request headers, stored in
- * #HTTPRequestHeaders_t.pBuffer, with initial configurations from 
+ * #HTTPRequestHeaders_t.pBuffer, with initial configurations from
  * #HTTPRequestInfo_t.
- * 
+ *
  * Upon return, #HTTPRequestHeaders_t.headersLen will be updated with the number
  * of bytes written.
- * 
+ *
  * TODO: Expand documentation.
- * 
+ *
  * @param[in] pRequestHeaders Request header buffer information.
  * @param[in] pRequestInfo Initial request header configurations.
  * @return #HTTP_SUCCESS if successful, an error code otherwise.
  * TODO: Update for exact error codes returned.
  */
-HTTPStatus_t HTTPClient_InitializeRequestHeaders( HTTPRequestHeaders_t * pRequestHeaders, 
-                                                  const HTTPRequestInfo_t * pRequestInfo );
+    HTTPStatus_t HTTPClient_InitializeRequestHeaders( HTTPRequestHeaders_t * pRequestHeaders,
```

Comment:
It is my intention that when someone finishes a function they also expand on the documentation.